packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -499,12 +499,24 @@ class DevFS {
 
     // Update modified files
     final Map<Uri, DevFSContent> dirtyEntries = <Uri, DevFSContent>{};
-
     int syncedBytes = 0;
+    if (fullRestart) {
+      generator.reset();
+    }
+    // On a full restart, or on an initial compile for the attach based workflow,
+    // this will produce a full dill. Subsequent invocations will produce incremental
+    // dill files that depend on the invalidated files.
+    _logger.printTrace('Compiling dart to kernel with ${invalidatedFiles.length} updated files');
+    final Future<CompilerOutput> pendingCompilerOutput = generator.recompile(
+      mainUri,
+      invalidatedFiles,
+      outputPath: dillOutputPath ?? getDefaultApplicationKernelPath(trackWidgetCreation: trackWidgetCreation),
+      packageConfig: packageConfig,
+    );
     if (bundle != null) {
-      final String assetBuildDirPrefix = _asUriPath(getAssetBuildDirectory());
       // The tool writes the assets into the AssetBundle working dir so that they
       // are in the same location in DevFS and the iOS simulator.
+      final String assetBuildDirPrefix = _asUriPath(getAssetBuildDirectory());
       final String assetDirectory = getAssetBuildDirectory();
       bundle.entries.forEach((String archivePath, DevFSContent content) {
```

Comment:
Where does the stat happen? Is it done as part of the `isModified` and `size` getters?

Reply:
Done